lib/unit-ruby/util/api_resource.rb

```diff
@@ -104,7 +116,7 @@ def self.belongs_to(resource_name, resource_type: nil, class_name: nil)
 
     # Hyrdates an instance of the resource from data returned from the API
     def self.build_resource_from_json_api(data_item)
-      new.tap do |resource|
+      response_resource.new.tap do |resource|
```

Comment:
(merging as-is since it's mostly just an ergonomics thing and works for our purposes rn, but happy to chat about better ways to do this in the future!)

Reply:
> i'm guessing stripe does something custom in the client when the action doesn't match nicely to HTTP verbs?

Yep, exactly. In Stripe's API, in addition to the 5 "standard" methods (Create, Retrieve, Update, Delete and List), a resource may have any number of custom methods. In API reviews we always pushed for custom methods to be named with verbs so that the invocation would feel natural in most languages, e.g. `payment_intent.confirm`: https://github.com/stripe/stripe-ruby/blob/a96e6f3b426b2d21566cc6bea0de800ed42eef0e/lib/stripe/resources/payment_intent.rb#L51-L58